### PR TITLE
Users availability query

### LINKED ISSUE #46: Add user's availability to get user query user type

### LINKED ISSUE #47: Store datetime (in availabilities and pairings) as UNIX time and convert to human readable format in endpoints

### LINKED ISSUE #53: Add FactoryBot + Faker gems

### PR DESCRIPTION
**Pull request by leahriffell**
### What was changed?
- Added availabilities to user type so that client can also fetch a user's availabilities 
- Added factorybot + faker and created a users and availabilities factory
  - Also added a way to create a user w/ existing availabilities
- Changed datetimes in schema (availabilities, pairings) into integer because we will store and send as unix time

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.

### Issues:
* Closes #46
* Closes #53 
* Closes #47 

### Tracking Consistency:
- [x] Added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion

### CODE REVIEW

File: app/graphql/types/user_type.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ class UserType < Types::BaseObject
       field :email, String, null: false
       field :created_at, GraphQL::Types::ISO8601DateTime, null: false
       field :updated_at, GraphQL::Types::ISO8601DateTime, null: false
+      field :availabilities, [Types::AvailabilityType], null: false
```
Comment (leahriffell) [inline app/graphql/types/user_type.rb:8]: Makes availabilities an added field that user can query

File: db/migrate/20201217034315_create_availabilities.rb [hunk 2]
```diff
@@ -1,9 +1,9 @@
 class CreateAvailabilities < ActiveRecord::Migration[6.1]
   def change
     create_table :availabilities do |t|
-      t.references :users, null: false, foreign_key: true
-      t.datetime :start_date_time
-      t.datetime :end_date_time
+      t.references :user, null: false, foreign_key: true
+      t.integer :start_date_time, :limit => 8
+      t.integer :end_date_time, :limit => 8
```
Comment (leahriffell) [inline db/migrate/20201217034315_create_availabilities.rb:6]: Was getting errors that some datetimes were too large so I increased the limit. The data type in schema is now a bigint.

File: db/migrate/20201219201758_change_availability_date_data_type.rb [hunk 3]
```diff
@@ -1,10 +0,0 @@
-class ChangeAvailabilityDateDataType < ActiveRecord::Migration[6.1]
-  def change
-    change_table :availabilities do |t|
-      t.remove :start_date_time, :end_date_time, :users_id
-      t.references :user, null: false, foreign_key: true
-      t.string :start_date_time
-      t.string :end_date_time
-    end
-  end
-end
```
Comment (leahriffell) [inline db/migrate/20201219201758_change_availability_date_data_type.rb:10]: Ended up changing some existing migrations so you may wanna reset your db (drop, create, migrate) 🙏🏻

File: spec/queries/users/get_user_spec.rb [hunk 4]
```diff
@@ -1,47 +1,53 @@
...
-                                  query($id: ID!) {
-                                      getUser(id: $id) {
-                                          id
-                                          username
-                                          email
-                                      }
-                                  }
-      GRAPHQL
+      @user_1 = create(:user)
+      @user_2 = create(:user, :with_availabilities)
```
Comment (leahriffell) [inline spec/queries/users/get_user_spec.rb:6]: You can use this syntax if you want to create a factorybot user that has 3 existing availabilities

**Review by lkriffell — APPROVED** [review]
Great work!

**Pull request merged**